llvm/lib/Target/X86/X86ISelLowering.cpp

```diff
@@ -32664,6 +32679,24 @@ static SDValue LowerPREFETCH(SDValue Op, const X86Subtarget &Subtarget,
   return Op;
 }
 
+static SDValue LowerFCanonicalize(SDValue Op, SelectionDAG &DAG) {
+  SDNode *N = Op.getNode();
+  SDValue Operand = N->getOperand(0);
+  EVT VT = Operand.getValueType();
+  SDLoc dl(N);
+
+  SDValue One = DAG.getConstantFP(1.0, dl, VT);
+
+  // TODO: Fix Crash for bf16 when generating strict_fmul as it
+  // leads to a error : SoftPromoteHalfResult #0: t11: bf16,ch = strict_fmul t0,
+  // ConstantFP:bf16<APFloat(16256)>, t5 LLVM ERROR: Do not know how to soft
+  // promote this operator's result!
+  SDValue Chain = DAG.getEntryNode();
+  SDValue StrictFmul = DAG.getNode(ISD::STRICT_FMUL, dl, {VT, MVT::Other},
```

Comment:
This should move to generic code as the default expansion

Reply:
@pawan-nirpal-031 Are you happy to work on a follow up patch?

Reply:
Yes I will address this in a follow up patch Simon. 

@arsenm won't it interfere with how the other targets want to handle it?

Reply:
No, it will only provide a reasonable default for other targets. Targets are still free to make it legal or custom lower as they choose

Reply:
Okay sure then. I will move it in the following PR, should I also handle constants in the next one or we keep it for later PR?

Reply:
PRs are ideally always as minimal as possible, so keep it separate

Reply:
Alright then, thanks for the suggestion. I will start creating the next PR.

Reply:
Hi @arsenm , should I do it in visitFCANONICALIZE in the generic combiner? I am not aware if there is any generic lowering!

Reply:
No. Usual place would be in LegalizeDAG, or a helper in TargetLowering used by LegalizeDAG